### PR TITLE
Add SCConv module to YOLO11

### PR DESCRIPTION
**Pull request by njmdin12**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Add a new SCConv residual convolution module and a minor docs formatting fix. 🚀

### 📊 Key Changes
- Introduce ultralytics/nn/modules/scconv.py defining an SCConv block (Conv-BN-ReLU → Conv-BN + residual → ReLU)
- Minor whitespace fix in docs/en/guides/hyperparameter-tuning.md

### 🎯 Purpose & Impact
- Provides a building block for experimenting with SCConv in YOLO11 architectures
- Not yet integrated into model configs or imports, so no user-facing changes until wired in
- Potential shape/stride risks: residual add will fail if in/out channels differ or stride>1; consider a projection/downsample path
- Requires tests, docs, and model/config wiring to realize benefits and ensure export/CI compatibility

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @username, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

Additional notes for this PR:
- Consider adding a projection/downsample path for the residual when stride>1 or in/out channels differ to avoid shape mismatches.
- Please wire SCConv into YOLO11 model configs and exports if intended, add imports where needed, and include unit tests covering forward pass and shape compatibility.

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). This is an automated message, and an engineer will be with you shortly to assist. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/nn/modules/scconv.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        self.conv2 = nn.Conv2d(out_channels, out_channels, kernel_size, stride=1, padding=1, groups=groups, bias=False)
+        self.bn2 = nn.BatchNorm2d(out_channels)
+        self.relu = nn.ReLU(inplace=True)
+
+    def forward(self, x):
+        identity = x
+        out = self.conv1(x)
+        out = self.bn1(out)
+        out = self.relu(out)
+        out = self.conv2(out)
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/modules/scconv.py:18]: ❗ **CRITICAL**: `out += identity` will raise whenever this block is instantiated with `stride != 1`, different `in_channels`/`out_channels`, or grouped convolutions that change the tensor shape. Please add a projection path (e.g. 1x1 conv) or at least guard/raise to keep the residual shapes aligned before the add.

File: ultralytics/nn/modules/scconv.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class SCConv(nn.Module):
+    def __init__(self, in_channels, out_channels, kernel_size=3, stride=1, padding=1, groups=1):
+        super(SCConv, self).__init__()
+        self.conv1 = nn.Conv2d(in_channels, out_channels, kernel_size, stride, padding, groups=groups, bias=False)
```
Comment (UltralyticsAssistant) [inline ultralytics/nn/modules/scconv.py:7]: ⚠️ **HIGH**: `self.conv2` hard-codes `padding=1`, so any call with `kernel_size` or `padding` different from 3/1 shrinks the feature map and breaks the residual add. Please reuse the incoming `padding` (or compute it from `kernel_size`) to keep the spatial dimensions consistent.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Please fix the residual path so the module either projects or enforces matching shapes, and ensure `conv2` uses the caller’s padding so custom kernel sizes don’t break the block.

💬 Posted 2 inline comments
